MatrixKit/Models/Account/MXKAccount.m

```diff
@@ -840,7 +840,7 @@ - (void)enablePusher:(BOOL)enabled success:(void (^)())success failure:(void (^)
                                @"url": self.pushGatewayURL,
                                };
     
-    NSString *deviceLang = [NSLocale preferredLanguages][0];
+    NSString *deviceLang = [[NSLocale currentLocale] objectForKey:NSLocaleLanguageCode];;
```

Comment:
VShell, the country code may be useful in language code to be able to differentiate for example en-GB and en-US.
I would prefer to keep the original code, and apply your patch only when the retrieved language code is longer than 8 characters. 

Note: please remove the double semicolon.